File: backend/src/main/java/com/mikelike/getpethelp/backend/domain/Worker.java
package com.mikelike.getpethelp.backend.domain;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name="worker")
public class Worker {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    Long id;
    @OneToOne(cascade = CascadeType.ALL,mappedBy="worker")
    private WorkerInfo workerInfo;
    private Double rating = 0.0;
    @OneToOne
    @JoinColumn(name = "user_id", nullable = false)
    private ShortUserInfo shortUserInfo;
    @Transient
    @ManyToMany(mappedBy = "workerInfoList")
    private List<Task> tasks = new ArrayList<>();
    @OneToMany(fetch = FetchType.LAZY, mappedBy = "worker")
    private List<Review> reviews = new ArrayList<Review>();
}

File: mobileClient/app/src/main/java/com/mikelike/getpethelp/mobile/domain/Worker.kt
package com.mikelike.getpethelp.mobile.domain

import com.mikelike.getpethelp.backend.domain.ShortUserInfo
import java.io.Serializable

data class Worker(
    var id: Long? = null,
    val workerInfo: WorkerInfo? = null,
    val rating: Double = 0.0,
    val shortUserInfo: ShortUserInfo? = null,
    val tasks: List<Task>? = null,
    val reviews: List<Review?> = ArrayList<Review>()
) : Serializable
